packages/core-mobile/app/new/features/buyOnramp/screens/SelectBuyAmountScreen.tsx

```diff
@@ -0,0 +1,326 @@
+import { ScrollScreen } from 'common/components/ScrollScreen'
+import React, { useCallback, useMemo, useState } from 'react'
+import {
+  Button,
+  View,
+  Text,
+  Icons,
+  Pressable,
+  useTheme,
+  FiatAmountInputWidget
+} from '@avalabs/k2-alpine'
+import { useRouter } from 'expo-router'
+import { LogoWithNetwork } from 'features/portfolio/assets/components/LogoWithNetwork'
+import { useErc20ContractTokens } from 'common/hooks/useErc20ContractTokens'
+import { useSearchableTokenList } from 'common/hooks/useSearchableTokenList'
+import { TokenUnit } from '@avalabs/core-utils-sdk'
+import { useNetworks } from 'hooks/networks/useNetworks'
+import { useWatchlist } from 'hooks/watchlist/useWatchlist'
+import { selectSelectedCurrency } from 'store/settings/currency'
+import { useSelector } from 'react-redux'
+import { useFormatCurrency } from 'common/hooks/useFormatCurrency'
+import { isTokenSupportedForBuying } from 'features/buyOnramp/utils'
+import { useOnRampServiceProvider, useOnRampToken } from '../store'
+import { useGetPurchaseLimits } from '../hooks/useGetPurchaseLimits'
+import { useSearchDefaultsByCountry } from '../hooks/useSearchDefaultsByCountry'
+import { useLocale } from '../hooks/useLocale'
+import { PaymentMethods, ServiceProviderCategories } from '../consts'
+
+export const SelectBuyAmountScreen = (): React.JSX.Element => {
+  const {
+    theme: { colors }
+  } = useTheme()
+  const { formatIntegerCurrency, formatCurrency } = useFormatCurrency()
+  const { getFromPopulatedNetwork } = useNetworks()
+  const [amount, setAmount] = useState(0)
+  const { navigate } = useRouter()
+  const selectedCurrency = useSelector(selectSelectedCurrency)
+  const { getMarketTokenBySymbol } = useWatchlist()
+  const [onrampToken] = useOnRampToken()
+  const [serviceProvider] = useOnRampServiceProvider()
+  const erc20ContractTokens = useErc20ContractTokens()
+  const { filteredTokenList } = useSearchableTokenList({
+    tokens: erc20ContractTokens,
+    hideZeroBalance: false
+  })
+  const { data: purchaseLimits } = useGetPurchaseLimits({
+    categories: [ServiceProviderCategories.CRYPTO_ONRAMP],
+    fiatCurrencies: [selectedCurrency],
+    cryptoCurrencyCodes: onrampToken?.currencyCode
+      ? [onrampToken?.currencyCode]
+      : undefined
+  })
+  const { countryCode } = useLocale()
+
+  const { data: defaultsByCountry } = useSearchDefaultsByCountry({
+    categories: [ServiceProviderCategories.CRYPTO_ONRAMP]
+  })
+
+  const defaultPaymentMethod = useMemo(() => {
+    const pm = defaultsByCountry?.find(d => d.countryCode === countryCode)
+      ?.defaultPaymentMethods[0]
+    return pm ? PaymentMethods[pm as keyof typeof PaymentMethods] : undefined
+  }, [defaultsByCountry, countryCode])
+
```

Comment:
there are so many hooks being used here. would be nicer to extract these into a hook to keep this UI component clean.